backend/server.js

```diff
@@ -24,6 +26,50 @@ import {
 } from './pubsub.js'
 import { addChannelToSubscription, deleteChannelFromSubscription, postEvent, pushServerActionhandlers, subscriptionInfoWrapper } from './push.js'
 
+const createWorker = (path: string): {
+  ready: Promise<void>,
+  rpcSbp: (...args: any) => Promise<any>,
+  terminate: () => Promise<number>
+} => {
+  const worker = new Worker(path)
+  const ready = new Promise((resolve, reject) => {
+    worker.on('message', (msg) => {
+      if (msg === 'ready') resolve()
+    })
+    worker.on('error', reject)
+  })
+
+  const rpcSbp_ = (...args: any) => {
+    return new Promise((resolve, reject) => {
+      const mc = new MessageChannel()
+      mc.port2.onmessage = (event) => {
+        const [success, result] = ((event.data: any): [boolean, any])
+        if (success) return resolve(result)
+        reject(result)
+      }
+      mc.port2.onmessageerror = () => {
+        reject(Error('Message error'))
+      }
+      worker.postMessage([mc.port1, ...args], [mc.port1])
+    })
+  }
+
+  let rpcSbp = (...args: any) => ready.then(() => rpcSbp_(...args))
+  // Avoid unncessary `ready.then` if we already know the promise has resolved
+  ready.then(() => {
+    rpcSbp = rpcSbp_
+  })
+
+  return {
+    ready,
+    rpcSbp,
+    terminate: () => worker.terminate()
+  }
+}
```

Comment:
Nicely designed and implemented. 👍 

I wonder if this should be put in some kind of utils.js file...

Reply:
Thanks! It's a simplified version of what we do for the SW.